### ISSUE TITLE
docker run -ti mesosphere/mesos-master:0.22.1-1.0.ubuntu1404 /bin/sh no such file or directory

### ISSUE THREAD

**Issue by cfortier2** [docker/docker#14754]
Hello, 

I'm able to reproduce this issue: https://github.com/docker/docker/issues/5135

Using CoreOS ami on AWS: CoreOS-alpha-681.0.0-hvm (ami-23b58613)

that gets upgraded to CoreOS 717.3.0

```
ip-10-43-3-56 ~ # cat /etc/os-release
NAME=CoreOS
ID=coreos
VERSION=717.3.0
VERSION_ID=717.3.0
BUILD_ID=
PRETTY_NAME="CoreOS 717.3.0"
ANSI_COLOR="1;32"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://github.com/coreos/bugs/issues"
```

Dockerfile is located at: https://github.com/behance/mesos-systemd/blob/master/v1/fleet_units/mesos-master.service
```
[Unit]
Description=MesosMaster
After=docker.service
Requires=docker.service

[Service]
Environment=MESOS_IMAGE=mesosphere/mesos-master:0.22.1-1.0.ubuntu1404
Environment=ZOOKEEPER=<ZK_ELB>

User=core
Restart=on-failure
RestartSec=20
TimeoutStartSec=0
ExecStartPre=-/usr/bin/docker kill mesos_master
ExecStartPre=-/usr/bin/docker rm mesos_master
ExecStartPre=/usr/bin/docker pull ${MESOS_IMAGE}
ExecStart=/usr/bin/sh -c "sudo /usr/bin/docker run \
    --name=mesos_master \
    --privileged \
    --net=host \
    -v /var/lib/mesos/master:/var/lib/mesos/master \
    ${MESOS_IMAGE} \
    --hostname=`curl -s http://169.254.169.254/latest/meta-data/public-hostname` \
    --log_dir=/var/log/mesos \
    --quorum=2 \
    --work_dir=/var/lib/mesos/master \
    --zk=zk://${ZOOKEEPER}/mesos"
ExecStop=/usr/bin/docker stop mesos_master
ExecStartPost=/usr/bin/docker pull cfortier/utility:latest

[Install]
WantedBy=multi-user.target

[X-Fleet]
Global=true
MachineMetadata=role=control
```

run command and error:
```
ip-10-43-3-56 ~ # docker run -ti mesosphere/mesos-master:0.22.1-1.0.ubuntu1404 /bin/sh
no such file or directory
FATA[0040] Error response from daemon: Cannot start container a533c4d1fc2107ca34f7189fd9ed49f3364fe823ab44a3f29c76d2a2b9d9600f: [8] System error: no such file or directory
```

**Comment by cfortier2** [comment]
```
ip-10-43-3-56 ~ # docker version
Client version: 1.6.2
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 7c8fca2-dirty
OS/Arch (client): linux/amd64
Server version: 1.6.2
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 7c8fca2-dirty
OS/Arch (server): linux/amd64
ip-10-43-3-56 ~ # docker info
Containers: 13
Images: 106
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Kernel Version: 4.0.5
Operating System: CoreOS 717.3.0
CPUs: 8
Total Memory: 14.69 GiB
Name: ip-10-43-3-56.us-west-2.compute.internal
ID: AK3G:ZORG:OXRS:UP43:4RD6:7ND2:YPPQ:UDFT:S5I7:DUAS:4YNJ:2HNI
ip-10-43-3-56 ~ # uname -a
Linux ip-10-43-3-56.us-west-2.compute.internal 4.0.5 #2 SMP Fri Jul 10 06:25:01 UTC 2015 x86_64 Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz GenuineIntel GNU/Linux
```

**Comment by phemmer** [comment]
Where are you seeing `exec: "/bin/sh": stat /bin/sh: no such file or directory`? I don't see that anywhere in the output you provided.

**Comment by cfortier2** [comment]
its the bottom of my initial post:

![image](https://cloud.githubusercontent.com/assets/2601345/8781575/e57afeda-2edd-11e5-8a83-a1b44f6afedb.png)

**Comment by phemmer** [comment]
That's a different issue than `exec: "/bin/sh": stat /bin/sh: no such file or directory`.

If this were a duplicate of the issue you linked, it would appear as:

    Error response from daemon: Cannot start container e06614e7332b: exec: "/bin/bash": stat /bin/bash: no such file or directory

**Comment by cfortier2** [comment]
ok, slight semantics. i renamed the issue.

**Comment by zhanghuanzhong** [comment]
when i run ```docker run -ti mesosphere/mesos-master:0.22.1-1.0.ubuntu1404 /bin/sh``` the container exit without any errors or output. so  i don't know  how to help you. :)

$ docker version
Client version: 1.3.2
Client API version: 1.15
Go version (client): go1.3.3
Git commit (client): 39fa2fa/1.3.2
OS/Arch (client): linux/amd64
Server version: 1.3.2
Server API version: 1.15
Go version (server): go1.3.3
Git commit (server): 39fa2fa/1.3.2

$ docker info
Containers: 7
Images: 57
Storage Driver: devicemapper
 Pool Name: docker-8:1-36045986-pool
 Pool Blocksize: 65.54 kB
 Data file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata file: /var/lib/docker/devicemapper/devicemapper/metadata
 Data Space Used: 7.169 GB
 Data Space Total: 1.074 TB
 Metadata Space Used: 10.78 MB
 Metadata Space Total: 2.147 GB
 Library Version: 1.02.89-RHEL6 (2014-09-01)
Execution Driver: native-0.2
Kernel Version: 2.6.32-431.5.1.el6.x86_64

**Comment by aidanhs** [comment]
The issue with `docker run -it` is expected:
```
$ docker history mesosphere/mesos-master:0.22.1-1.0.ubuntu1404
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
0248bf46706b        12 weeks ago        /bin/sh -c #(nop) ENTRYPOINT ["mesos-master"]   0 B                 
7ca9fcacd58d        12 weeks ago        /bin/sh -c #(nop) CMD ["--registry=in_memory"   0 B                 
44b14f43a9d2        12 weeks ago        /bin/sh -c #(nop) MAINTAINER Mesosphere <supp   0 B                 
eb96c4a15f9a        12 weeks ago        /bin/sh -c apt-get -y install mesos=0.22.1-1.   295 MB              
9f956ecd28fc        12 weeks ago        /bin/sh -c apt-get -y update                    20.78 MB            
e9b6898b231c        12 weeks ago        /bin/sh -c apt-key adv --keyserver keyserver.   25.84 kB            
51b6a2e302b0        12 weeks ago        /bin/sh -c echo "deb http://repos.mesosphere.   51 B                
822d24b62c0a        12 weeks ago        /bin/sh -c #(nop) MAINTAINER Mesosphere <supp   0 B                 
2d24f826cb16        5 months ago        /bin/sh -c #(nop) CMD [/bin/bash]               0 B                 
117ee323aaa9        5 months ago        /bin/sh -c sed -i 's/^#\s*\(deb.*universe\)$/   1.895 kB            
1c8294cc5160        5 months ago        /bin/sh -c echo '#!/bin/sh' > /usr/sbin/polic   194.5 kB            
fa4fd76b09ce        5 months ago        /bin/sh -c #(nop) ADD file:0018ff77d038472f52   188.1 MB            
511136ea3c5a        2 years ago                                                         0 B                 Imported from -
```

You need to override the entrypoint if you want to start a shell. Have a read of https://docs.docker.com/reference/run/#entrypoint-default-command-to-execute-at-runtime

The .service file you linked is not a dockerfile and I'm not sure what relevance that has?

**Comment by cpuguy83** [comment]
Yes, this error can be coming from the entrypoint wanting to take arguments like that.
It could also be from a known error around removing layers that shouldn't be removed, which is fixed in docker 1.8.x
Is this still a problem for you?
You'd likely have to remove the image and re-pull.

**Comment by cfortier2** [comment]
we finally figured it out that we were pulling the mesos-master, marathon, and chronos containers at the same time. We changed the order of operations to pull them serially and haven't seen an issue since then.

Thanks!
